client/dashboard/utils/site-plan.ts

```diff
@@ -85,5 +90,11 @@ export function getSitePlanDisplayName( site: Site ) {
 		}
 	}
 
-	return site.plan?.product_name || site.plan?.product_name_short || '';
+	// Display the short name for WP.com plans.
+	// Determine if the plan is a WP.com plan by checking if the license key is empty.
+	if ( ! plan.license_key ) {
+		return plan.product_name_short || '';
```

Comment:
Do we still need `|| ''` here and in line 99? I think they are non-nullable fields.